io/io/test/TFileMergerTests.cxx

```diff
@@ -0,0 +1,48 @@
+#include "TFileMerger.h"
+
+#include "TMemFile.h"
+#include "TTree.h"
+
+#include "gtest/gtest.h"
+
+static void CreateATuple(TMemFile &file, const char *name, double value)
+{
+   auto mytree = new TTree(name, "A tree");
+   mytree->Branch(name, &value);
+   mytree->Fill();
+   file.Write();
+}
+
+static void CheckTree(TMemFile &file, const char *name, double expectedValue)
+{
+   auto t = static_cast<TTree *>(file.Get(name));
+   ASSERT_TRUE(t != nullptr);
+
+   double d;
+   t->SetBranchAddress(name, &d);
+   t->GetEntry(0);
+   EXPECT_EQ(expectedValue, d);
+}
+
+TEST(TFileMerger, CreateWithTFilePointer)
+{
+   TMemFile a("a.root", "CREATE");
+   CreateATuple(a, "a_tree", 1.);
+
+   // FIXME: Calling this out of order causes two values to be written to the second file.
+   TMemFile b("b.root", "CREATE");
+   CreateATuple(b, "b_tree", 2.);
+
+   TFileMerger merger;
+   auto output = new TMemFile("output.root", "CREATE");
```

Comment:
Why not having output being a unique_ptr? (Maybe using std::make_unique :) )

Reply:
IIRC, the TMemFile got deleted by the `merger`.

Reply:
Yes, but then you do:
```
 merger.OutputFile(std::unique_ptr<TFile>(output));
```
So I don't see the difference :) (and yes ownership needs to pass for this function to the TMemFile ....